### ISSUE TITLE
Default persmissions to permissive ?

### ISSUE THREAD

**Issue by dbeniamine** [Runalyze/Runalyze#1876]
Hi,

I realized that by default every file of runalyze is readable by any one (`644`, `755` for directories) which seems not very safe as `data/config.php` contains in clear the DB password. A rogue user could read this file and get access to runalyze DB ...

I guess that a setting the ownership of runalyze to `root:www-data`, and each files permission to `0640` (`0750` for directories) by default, and `0660` (`0770`) for the part that must be writable for runalyze to work (`data/log`, `data/cache`, `data/import`, `data/backup-tool/import`, `data/backup-tool/backup`) would be more secure.

There is a small script that set these values (assuming that it is in the runalyze directory), I think it could help to include it in the distribution and recommand to run it after install and upgrade:

    #!/bin/bash
    runalyzedir=$(dirname $0)
    www_user="www-data"
    WRITABLE_PATH="data/cache data/log data/import data/backup-tool/import data/backup-tool/backup data/config.php"

    # $1 directory
    # $2 perm for file
    # $3 perm for subdirectories
    set_perms(){
        find $1 -type f -exec chmod $2 {} \;
        find $1 -type d -exec chmod $3 {} \;
    }

    echo "Setting default ownership"
    chown -R root:$www_user $runalyzedir
    echo "Setting default permissions"
    set_perms $runalyzedir 0640 0750

    for wp in $WRITABLE_PATH
    do
        echo "Making $wp writable"
        set_perms $runalyzedir/$wp 0660 0770
    done

    echo "Restoring execution permissions for scripts"
    find $runalyzedir -name *.sh -o -name *.pl -exec chmod gu+x {} \;

David

**Comment by mipapo** [comment]
In general you are right. With the next RUNALYE version 3.0 many things will change regarding some directories.
​
In general we could chmod the directories in git. Just have to check that the permissions not changing when i do the release packages. So I don't think we should add this script as most of the users can't execute any scripts...For most of the people it will be to complicated.

**Comment by dbeniamine** [comment]
I agree, but I though that zip does not preserve permissions, and  issue https://github.com/Runalyze/Runalyze/issues/1874 let me thing that most users relies on the zip archive hence the script.

**Comment by mipapo** [comment]
The users who use the zip files won't have the skills or permissions to execute such a bash script...

**Comment by dbeniamine** [comment]
Fair enough.

**Comment by mipapo** [comment]
But we'll have a look into the permissions and so on with symfony
